core/snippets/snippets_parser.py

```diff
@@ -340,7 +343,12 @@ def parse_body(text: str) -> Union[str, None]:
     if match_leading is None:
         return None
 
-    return text[match_leading.start() :].rstrip()
+    body = text[match_leading.start() :].rstrip()
+
+    if ESCAPED_SNIPPET_DELIMITER_EXPRESSION.search(body):
```

Comment:
I don't really see a point in doing a search follow by a substitution. I would just do the substitution directly.